internal/controllers/extension_controller.go

```diff
@@ -189,7 +179,12 @@ func (r *ExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alpha1.Ext
 		return ctrl.Result{}, nil
 	}
 
-	app := r.GenerateExpectedApp(*ext, bundle.Image)
+	bundleVersion, err := bundle.Version()
+	if err != nil {
+		l.Info("could not determine bundle version", "name", ext.GetName(), "namespace", ext.GetNamespace())
+	}
+
+	app := r.GenerateExpectedApp(*ext, bundle.Image, bundleVersion.String())
```

Comment:
will we get an NPE on `bundleVersion.String()` here if `bundle.Version()` returned an error?

Reply:
Also, should we just pass the `bundle` to `GenerateExpectedApp` and grab the version inside? Simplifies the function signature and makes the bundle available to the function in case other bundle attributes are needed.

Reply:
+1, will add

Reply:
Updated